instrumentation/openai-client-instrumentation/common/src/main/java/co/elastic/otel/openai/wrappers/ChatCompletionEventsHelper.java

```diff
@@ -102,7 +115,8 @@ public static void emitPromptLogEvents(
           bodyBuilder.put("id", toolMsg.toolCallId());
         }
       } else {
-        throw new IllegalStateException("Unhandled type : " + msg.getClass().getName());
+        LOG.log(Level.WARNING, "Unhandled OpenAI message type will be dropped: {0}", msg);
```

Comment:
This would previously cause the instrumentation to fail, which is a bit too strict.